From e83741d23e1979b8d395862b41c54a2187dffd55 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:34:01 -0400
Subject: [alsa] Revert: hda: drop def association and sequence from pinconf comparing

Message-id: <140972823573-319-patchset-jkysela@redhat.com>
Patchwork-id: 91015
O-Subject: [RHEL 7.1 PATCH 318/442] ALSA: Revert "ALSA: hda - drop def association and sequence from pinconf comparing"
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

This reverts commit c687200b9d4ef60042a50f7d942cfef120cc7bf1.

Dropping the def association and sequence from pinconf comparing is a
bit risky, It will introduce a greater risk of catching unwanted
machines.

And in addition, so far no BIOS experts give us an explicit answer
whether it makes senses to compare these two fields or not.

For safety reason, we revert this commit.

Signed-off-by: Hui Wang <hui.wang@canonical.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 37df09492c3315c4824e53ebcea781c114be9f41)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/hda_auto_parser.c b/sound/pci/hda/hda_auto_parser.c
index 3cf9137..b684c6e 100644
--- a/sound/pci/hda/hda_auto_parser.c
+++ b/sound/pci/hda/hda_auto_parser.c
@@ -844,8 +844,7 @@ static bool pin_config_match(struct hda_codec *codec,
 {
  for (; pins->nid; pins++) {
   u32 def_conf = snd_hda_codec_get_pincfg(codec, pins->nid);
-  u32 mask = 0xffffff00;
-  if ((pins->val & mask) != (def_conf & mask))
+  if (pins->val != def_conf)
    return false;
  }
  return true;
-- 
1.7.1